databases/functions/users/get-blocked-and-blocker-user-id.js

```diff
@@ -0,0 +1,19 @@
+const {Op} = require('sequelize');
+
+const getBlockedAndBlockerUserId = async (userBlockUserModel, userId) => {
+  if (!userId) throw new Error('User id is required');
+  const users = await userBlockUserModel.findAll({
+    where: {
+      [Op.or]: [{user_id_blocker: userId}, {user_id_blocked: userId}]
+    },
+    raw: true
+  });
+  let userIds = new Set([]);
+  for (const user of users) {
+    userIds.add(user.user_id_blocked);
+    userIds.add(user.user_id_blocker);
+  }
+  return Array.from(userIds);
+};
+
+module.exports = getBlockedAndBlockerUserId;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getBlockedAndBlockerUserId` seems to be working as expected. However, it's adding both the blocker and blocked user IDs to the set `userIds`. This might lead to unexpected results if a user is both blocking and being blocked by different users. The function would return the user's ID in the list of blocked/blocking users, which could cause issues in the calling functions.

To fix this, you should only add the `user_id_blocked` when the `user_id_blocker` equals the input `userId`, and vice versa. Here's how you can modify your code:

```diff
  for (const user of users) {
    if(user.user_id_blocker === userId) {
-       userIds.add(user.user_id_blocked);
+       userIds.add(user.user_id_blocked);
    }
    if(user.user_id_blocked === userId) {
-       userIds.add(user.user_id_blocker);
+       userIds.add(user.user_id_blocker);
    }
  }
```
This way, you're ensuring that you're only getting the IDs of users who are either blocked by the given user or blocking the given user, not including the user themselves.


<!-- This is an auto-generated comment by OSS CodeRabbit -->